src/__private_api.rs

```diff
@@ -58,22 +59,30 @@ fn log_impl(
     #[cfg(feature = "kv")]
     builder.key_values(&kvs);
 
-    crate::logger().log(&builder.build());
+    logger.log(&builder.build());
 }
 
-pub fn log<'a, K>(
+pub fn log<'a, K, L>(
+    logger: &L,
```

Comment:
For binary size, I think the type of logger should be `L` instead of `&L`, so that it allows us to pass zero-sized logger. A default logger could be defined as as a zero-sized type which forwards logging calls to `create::logger()`, just like what the allocator [`Global`](https://doc.rust-lang.org/stable/std/alloc/struct.Global.html) does.

Reply:
Sounds reasonable. We have:

```rust
impl<T> Log for &'_ T { ... }
```

Let me send an updater later today.

Reply:
@EFanZh This is resolved at https://github.com/rust-lang/log/pull/664/commits/cde258593221716c7b64c049d6619b49b30dd4de now. Please check if it's what you expect.